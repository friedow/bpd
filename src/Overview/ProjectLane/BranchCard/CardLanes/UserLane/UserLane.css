.user-card-lane {
  color: white;
  padding: 5px 24px 10px 24px;
}

.user-card-lane .chip {
  margin-bottom: 0px;
}
