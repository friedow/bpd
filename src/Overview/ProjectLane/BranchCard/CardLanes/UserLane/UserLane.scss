.user-card-lane {
  color: white;
  align-items: center;
  display: flex;
  padding: 3px;
}

.user-card-lane .user-image {
  height: 100%;
}
.user-card-lane .user-name {
  padding: 0px 10px;
}
