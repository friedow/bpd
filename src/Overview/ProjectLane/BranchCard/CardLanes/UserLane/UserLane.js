import React, { Component } from 'react';
import './UserLane.css';

class UserLane extends Component {
  constructor(props) {
    super(props);
    this.state = {username: this.props.username,
        avatarUrl: "https://upload.wikimedia.org/wikipedia/commons/d/d2/Blank.png"};
  }
  componentDidMount() {
    if(this.props.avatarUrl) {
      this.setState({avatarUrl: this.props.avatarUrl});
    }
  }

  render() {
    return (
      <div className="card-lane user-card-lane">
        <div className="chip"><img src={this.state.avatarUrl} />{this.state.username}</div>
      </div>
    );
  }
}

export default UserLane;
