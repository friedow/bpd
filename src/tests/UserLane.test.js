import React from 'react';
import UserLane from '../Overview/ProjectLane/BranchCard/CardLanes/UserLane/UserLane.js';
import renderer from 'react-test-renderer';

it('UserLane renders correctly', () => {
    const tree = renderer
        .create(<UserLane
            username="John Doe"
            lastCommitTime="23:59"
            avatarUrl="http://pictures-of-john-doe.com/profile.png"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

