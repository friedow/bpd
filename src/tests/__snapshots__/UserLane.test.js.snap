exports[`test UserLane renders correctly 1`] = `
<div
  className="card-lane user-card-lane grey darken-3">
  <img
    className="user-image"
    src="http://pictures-of-john-doe.com/profile.png" />
  <p
    className="user-name">
    John Doe
  </p>
  <p
    className="last-committed">
    23:59
  </p>
</div>
`;
